masonry/src/passes/event.rs

```diff
@@ -195,6 +195,25 @@ pub(crate) fn root_on_access_event(
         event,
         false,
         |widget, ctx, event| {
+            if event.target == ctx.widget_id() {
+                match event.action {
+                    accesskit::Action::Focus => {
```

Comment:
Ideally, I'd like this whole block to be split out into a separate pass, to avoid having logic branches inside pass branch like we did with WidgetPod.

It's not a blocking concern. This code is fine for now, though I'd want to add a TODO comment.
```suggestion
            // TODO - Split into "access_event_focus" pass or something similar.
            if event.target == ctx.widget_id() {
                match event.action {
                    accesskit::Action::Focus => {
```

Reply:
In its current state, I agree. This behaviour makes it really hard to have custom handling of the Focus/Blur events anyway, so a hyper-targetted pass makes sense.
That would also be `O(1)` in the blur case, as it could just check if the target of the blur is focused (although in the focus case you need to re-confirm that it is focusable (in case of race conditions?))

See also https://github.com/linebender/xilem/pull/596#discussion_r1764998676 - if we had that, a separate pass would be invalid.

Reply:
To be clear, that's emphatically not needed to land this pr, and I'd prefer to land it with just this comment as-is